### ISSUE TITLE
The code editor is not accessible to screen readers

### ISSUE THREAD

**Issue by pamelafox** [Khan/live-editor#244]
A blind student is attempting to learn JS and running into this issue, which is also reported at ACE:
https://github.com/ajaxorg/ace/issues/2164

Hopefully the ACE team will be able to tackle this, but I'm filing here so it stays on the radar.

**Comment by kevinb7** [comment]
I was curious to hear/see what the editor would sound like with a screen reader so I did some Googling and found this Chrome extension: http://www.chromevox.com/.  This particular screen reader will read the first line of code.  I think this is going to be a tough one.  We'd probably want to have "group" or "region" roles for each block of code so users can navigate quickly.  

The biggest issue I see is that ACE puts each line in its div, but the structure of the code necessitates a structure that may split lines and/or span multiple lines.  It may be possible to have a div that contains the program structure along with aria attributes.  If we make that div content editable and forward all events to ACE the screen reader should worker while maintaining editor's current operations.  As ACE updates the cursor/selection we can use that to update the cursor/selection in the screen reading div.
